### PR TITLE
RR-738 Continue switch to short question set

### PR DESCRIPTION
**Pull request by mattwills-moj-digital**
This PR progresses the journey from switching from the "long" question set to the "short" one (i.e. when the Prisoner is no longer looking to work after release). It gets as far as submitting the penultimate page (the `/in-prison-training` view), but doesn't progress any further because the "Check Your Answers" page has not yet been incorporated into the flow.

The bulk of this PR is the addition of new unit tests, but there are some changes to functionality, particularly in relation to the back links.

### CODE REVIEW

File: server/routes/induction/common/qualificationDetailsController.ts [hunk 1]
```diff
@@ -18,7 +18,13 @@ export default abstract class QualificationDetailsController extends InductionCo
     if (!pageFlowHistory) {
       return res.redirect(`/plan/${prisonerSummary.prisonNumber}/view/education-and-training`)
     }
+
     const { prisonNumber } = req.params
+    if (!qualificationLevelForm) {
+      // Guard against the user using the back button to return to this page, which can cause a NPE on line 40 (depending on which pages they've been to)
+      // TODO - not sure if this is the best solution?
```
[the remark this answers is not in the document]
Comment (nathanrussell-moj-digital) [reply server/routes/induction/common/qualificationDetailsController.ts:25]: I'm sure what you've done is right, but I can't quite follow the scenario from the diff and comment.
I'm sure it's fine, and I'm sure we'll cover it in our hand over later 👍

File: server/routes/induction/dynamicAriaTextResolver.ts [hunk 2]
```diff
@@ -18,8 +18,12 @@ const getDynamicBackLinkAriaText = (req: Request, backLinkUrl: string): string =
     '/plan/{PRISON_NUMBER}/view/education-and-training': `Back to ${prisonerName}'s learning and work progress`,
     '/plan/{PRISON_NUMBER}/view/work-and-interests': `Back to ${prisonerName}'s learning and work progress`,
     '/prisoners/{PRISON_NUMBER}/induction/hoping-to-work-on-release': `Back to Is ${prisonerName} hoping to get work when they're released?`,
+    '/prisoners/{PRISON_NUMBER}/induction/reasons-not-to-get-work': `Back to What could stop ${prisonerName} working when they are released?`,
     '/prisoners/{PRISON_NUMBER}/induction/highest-level-of-education': `Back to What's the highest level of education ${prisonerName} completed before entering prison?`,
     '/prisoners/{PRISON_NUMBER}/induction/qualifications': `Back to ${prisonerName}'s qualifications`,
+    '/prisoners/{PRISON_NUMBER}/induction/additional-training': `Back to Does ${prisonerName} have any other training or vocational qualifications?`,
+    '/prisoners/{PRISON_NUMBER}/induction/in-prison-work': `What type of work would ${prisonerName} like to do in prison?`,
```
Comment (nathanrussell-moj-digital) [inline server/routes/induction/dynamicAriaTextResolver.ts:25]: Minor, should this be prefixed with "Back to" ?, or is this some weird edge case where the CIAG UI does it differently without the prefix?

File: server/routes/induction/update/additionalTrainingUpdateController.test.ts [hunk 3]
```diff
@@ -173,7 +170,6 @@ describe('additionalTrainingUpdateController', () => {
 
       // Then
       expect(res.render).toHaveBeenCalledWith('pages/induction/additionalTraining/index', expectedView)
-      expect(req.session.additionalTrainingForm).toBeUndefined()
```
Comment (nathanrussell-moj-digital) [inline server/routes/induction/update/additionalTrainingUpdateController.test.ts:176]: Do we no longer expect the form to be undefined? I can't see that the behaviour/code in the implementation has changed in this respect? 🤔

File: server/routes/induction/dynamicAriaTextResolver.ts (hunk 2 above)
Comment (mattwills-moj-digital) [reply server/routes/induction/dynamicAriaTextResolver.ts:25]: Oops, good spot. Fixed.

File: server/routes/induction/update/additionalTrainingUpdateController.test.ts (hunk 3 above)
Comment (nathanrussell-moj-digital) [reply server/routes/induction/update/additionalTrainingUpdateController.test.ts:176]: Ah, OK 👍

**Review by nathanrussell-moj-digital — APPROVED** [review]
LGTM 👍
